lib/rules/no-compare-neg-zero.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview The rule should warn against code that tries to compare against -0.
+ * @author no-compare-neg-zero
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow compare against -0.",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+
+        //--------------------------------------------------------------------------
+        // Helpers
+        //--------------------------------------------------------------------------
+
+        /**
+         * Checks a given node is -0
+         *
+         * @param {ASTNode} node - A node to check.
+         * @returns {boolean} `true` if the node is -0.
+         */
+        function isNegZero(node) {
+            return node.type === "UnaryExpression" && node.operator === "-" && node.argument.type === "Literal" && node.argument.value === 0;
+        }
+
+        return {
+            BinaryExpression(node) {
+                if (/^(?:[<>]|[!=]=)=?$/.test(node.operator)) {
+                    if (isNegZero(node.left) || isNegZero(node.right)) {
+                        context.report({
+                            node,
+                            message: "disallow use the {{operator}} operator to compare against -0.",
```

Comment:
@Aladdin-ADD I think the message is fine even when the operands are reversed. Equality is symmetric (`a == b` implies `b == a`) after all! :smile:

For the other comparison operators, I think it's more important that we show the operator the user is actually using, so we don't need to modify it in the error message. So the message should work for those operators as well.